src/core/public/application/application_service.tsx

```diff
@@ -306,6 +306,7 @@ export class ApplicationService {
         const relUrl = http.basePath.prepend(getAppUrl(availableMounters, appId, path));
         return absolute ? relativeToAbsolute(relUrl) : relUrl;
       },
+      parseAppUrl: (url) => parseAppUrl(url, http.basePath, this.apps),
```

Comment:
As the new ``RedirectCrossAppLinks` component should ignore links pointing to the current application, I couldn't just use `navigateToApp` directly, and had to expose `parseAppUrl` to allow the consumer to check if the app from the link is, or not, the active app.

Exposing this API still makes sense anyway I think